### DISCUSSION TITLE
How can I rename a Unique Index in migration.sql without deleting my whole data?

### DISCUSSION THREAD

**Discussion by createdbyken** [prisma/prisma#9239] [Q&A] [answered]
Hello! I'm facing quite an issue here. I am able to create migrations perfectly, but the problem that I'm having is that when the migration has been runner with `npx prisma run dev` this creates a Unique Index, which is good and bad. 

## How is created

CREATE UNIQUE INDEX "Post.image_unique" ON "Post"("image");

## How I need it

CREATE UNIQUE INDEX "Post_image_unique" ON "Post"("image");

I need to find a way to replace the "." for the "_"

So, when I edit my migration.sql file and run `npx prisma run dev` it says something like:

```
The following is a summary of the differences between the expected database schema given your migrations files, and the actual schema of the database.

It should be understood as the set of changes to get from the expected schema to the actual schema.

[*] Changed the  Post table
  [*] Renamed index Post_image_unique to Post.image_unique

We need to reset the PostgreSQL database "app_test" at "localhost:5433".
Do you want to continue? All data will be lost. › (y/N)
```

I cannot delete each time my data when running this. So my question is, how can rename my unique index?
I would appreciate the help!

**Comment by ryands17** [comment, marked as answer]
@createdbyken 👋 

You can use named constraints for this to add the name directly to the schema: https://www.prisma.io/docs/concepts/components/prisma-schema/names-in-underlying-database#constraint-and-index-names.

**Reply by createdbyken** [reply]
Thanks this actually fixed the problem!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
